plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -0,0 +1,221 @@
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.utils.common.CommandLineTool
+
+import org.semver4j.RangesList
+import org.semver4j.RangesListFactory
+
+import retrofit2.HttpException
+
+private const val LOCKFILE_NAME = "MODULE.bazel.lock"
+
+private val JSON = Json {
+    ignoreUnknownKeys = true
+}
+
+class Bazel(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analysisRoot, analyzerConfig, repoConfig), CommandLineTool {
+    class Factory : AbstractPackageManagerFactory<Bazel>("Bazel") {
+        override val globsForDefinitionFiles = listOf("MODULE", "MODULE.bazel")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Bazel(type, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    override fun command(workingDir: File?) = "bazel"
+
+    override fun getVersionArguments() = "version"
+
+    override fun transformVersion(output: String) = transformBazelVersion(output)
+
+    // Bazel 6.0 already supports bzlmod but it is not enabled by default.
+    // Supporting it would require adding the flag "--enable_bzlmod=true" at the correct position of all bazel
+    // invocations.
+    override fun getVersionRequirement(): RangesList = RangesListFactory.create(">=7.0")
+
+    override fun resolveDependencies(definitionFile: File, labels: Map<String, String>): List<ProjectAnalyzerResult> {
+        val projectDir = definitionFile.parentFile
+        val lockFile = projectDir.resolve(LOCKFILE_NAME)
+        val projectVcs = processProjectVcs(projectDir)
+        val moduleMetadata = Parser(definitionFile.readText()).parse()
+        val depDirectives = moduleMetadata.dependencies.associateBy { "${it.name}@${it.version}" }
+        val scopes = getDependencyGraph(projectDir, depDirectives)
+
+        // If no lock file is present, getDependencyGraph() runs "bazel mod graph", which creates a MODULE.bazel.lock
+        // file as a side effect. That file contains the URL of the Bazel module registry that was used for dependency
+        // resolution.
+        val lockFileData = parseLockfile(lockFile)
+
+        val bmrClient = BazelModuleRegistryClient.create(lockFileData.registryUrl())
+        val packages = scopes.flatMap { getPackages(it.dependencies, bmrClient) }.filterNotNull()
+
+        return listOf(
+            ProjectAnalyzerResult(
+                project = Project(
+                    id = Identifier(
+                        type = managerName,
+                        name = moduleMetadata.module?.name ?: "root",
```

Comment:
As per `OrtResult` invariant the identifier is unique. In particular the fallback to `root` is problematic in this regard. In other places IIRC we utilize the definition file path to come to something unique.

problematic case: multiple definition files